### ISSUE TITLE
Correct exports of the module for better module loading compatibility with node.js

### ISSUE THREAD

**Issue by trygve-lie** [googlemaps/js-markerclusterer#908]
**Is your feature request related to a problem? Please describe.**

This module exports a UMD module in the `main` field and it provides an ESM version through the `module` field in package.json.  Node.js supports mixing CJS and ESM modules these days but this setup does not go well with using this module in node.js.

The `module` field in node.js are only used by bundlers and not node.js itself. So when trying to use this module with node.js without bundling it will try to load the file references in the `main` field which is a UMD file that node.js does not understand.

Iow; how the ESM version is exposed in this module is wrong according to node.js.

My problem with this is that this breaks SSR of Lit web components (I am using this module in a Lit web component) and it also breaks [SSR in Nuxt](https://github.com/googlemaps/js-markerclusterer/issues/885) it seems like. Fixing the export of ESM will probably also [fix this issue in Angular](https://github.com/googlemaps/js-markerclusterer/issues/610).

 **Describe the solution you'd like**

Set package.json to `type: module` and export the ESM as recommended by node.js: https://nodejs.org/api/packages.html#dual-commonjses-module-packages

Basically add:

```json
  "type": "module",
  "exports": {
    "import": "./dist/index.esm.js",
    "require": "./dist/index.umd.js"
  },
```

 **Describe alternatives you've considered**

I don't think there is an alternative way of doing this.

**Comment by renaudcollet** [comment]
I import the package like this, so I can build and preview my Nuxt project : 
`import * as MarkerClustererPackage from '@googlemaps/markerclusterer';`
`let markerCluster: MarkerClustererPackage.MarkerClusterer = new MarkerClustererPackage.MarkerClusterer({...});`
